### ISSUE TITLE
Unable to pass log_retention_days as integer variable while using log-group resource and passing variable from accounts.yml in c7n-org

### ISSUE THREAD

**Issue by ShivaSharm** [cloud-custodian/cloud-custodian#8794]
### Describe the bug

Using 0.9.29.0 version for custodian, Trying to set up policy using c7n-org to set log retention for log groups and passing retention days as variable from accounts.yml config file:

```yaml
policies:
  - name: cloudwatch-logs-retention
    resource: log-group
    description: |
      Sets a default retention on a cloudwatch log group where a retention is currently unset.
    filters:
      - type: value
        key: retentionInDays
        value: absent
    actions:
      - type: retention
        days: "{cw_log_retention_days}"
```

accounts.yml
```yaml
accounts:
    - account_id: "XXXXXXXXXX"
      name: "account-alias"
      role: arn:aws:iam::XXXXXXXX:role/custodian-role
      vars:
         cw_log_retention_days: 15
```

Error: 

2023-08-01 09:11:53,219: custodian.commands:ERROR             Error on policy:cloudwatch-logs-retention resource:log-group
'{cw_log_retention_days}' is not of type 'integer'

            Failed validating 'type' in schema[12]['properties']['days']:
                {'type': 'integer'}

            On instance['days']:
                '{cw_log_retention_days}'

Does not work even if I try as below in actions for policy, and it is being rendered as None.
actions:
      - type: retention
        days: {cw_log_retention_days}

error:
2023-08-01 09:14:19,885: custodian.commands:ERROR             Error on policy:cloudwatch-logs-retention resource:log-group
{'cw_log_retention_days': None} is not of type 'integer'

            Failed validating 'type' in schema[12]['properties']['days']:
                {'type': 'integer'}

            On instance['days']:
                {'cw_log_retention_days': None}

### What did you expect to happen?

Is there a way to be able to pass an integer variable from accounts.yml without it being rendered as string. I expect we should be able to pass integer values as well for variables.

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

```shell
/ # custodian version --debug

Please copy/paste the following info along with any bug reports:

Custodian:   0.9.29
Python:      3.11.4 (main, Jun  9 2023, 02:29:05) [GCC 12.2.1 20220924]
Platform:    posix.uname_result(sysname='Linux', nodename='71ec8abd9250', release='5.10.16.3-microsoft-standard-WSL2', version='#1 SMP Fri Apr 2 22:23:49 UTC 2021', machine='x86_64')
Using venv:  False
Docker: True
Installed:

argcomplete==3.1.1
attrs==23.1.0
boto3==1.28.5
botocore==1.31.5
docutils==0.18.1
importlib-metadata==5.2.0
jmespath==1.0.1
jsonschema==4.17.3
pyrsistent==0.19.3
python-dateutil==2.8.2
pyyaml==6.0.1
s3transfer==0.6.1
six==1.16.0
tabulate==0.9.0
typing-extensions==4.7.1
urllib3==1.26.16
zipp==3.15.0
```

### Policy

```shell
policies:
  - name: cloudwatch-logs-retention
    resource: log-group
    description: |
      Sets a default retention on a cloudwatch log group where a retention is currently unset.
    filters:
      - type: value
        key: retentionInDays
        value: absent
    actions:
      - type: retention
        days: "{cw_log_retention_days}"


accounts.yml

accounts:
  - account_id: "XXXXXXXXXX"
    name: "account-alias"
    role: arn:aws:iam::XXXXXXXX:role/custodian-role
    vars:
      cw_log_retention_days: 15
```

### Relevant log/traceback output

```shell
2023-08-01 09:11:53,219: custodian.commands:ERROR             Error on policy:cloudwatch-logs-retention resource:log-group
'{cw_log_retention_days}' is not of type 'integer'

            Failed validating 'type' in schema[12]['properties']['days']:
                {'type': 'integer'}

            On instance['days']:
                '{cw_log_retention_days}'
```

**Comment by kapilt** [comment]
that's odd I was able to run your example, I did get an error that retention values had to be a particular range of values.

```
Exception running policy:cloudwatch-logs-retention account:Sandbox-Kapil region:us-east-1 error:An error occurred (InvalidParameterException) when calling the PutRetentionPolicy operation: Invalid retention value. Valid values are: [1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1096, 1827, 2192, 2557, 2922, 3288, 3653]
```

but changing the variable to 14 worked fine.

```
❯ c7n-org run -c org.yaml -u policy.yaml -s out
2023-08-01 18:45:59,420: c7n_org:INFO Policy resource counts Counter({'cloudwatch-logs-retention': 7})
```


is it possible you have multiple versions of custodian and c7n-org installed? 
`which c7n-org` should match directory on `which c7n`
and `pip freeze | grep c7n_org` should say `0.6.29`


this sounds like an issue that was fixed last year re 
https://github.com/cloud-custodian/cloud-custodian/pull/7832

**Comment by ShivaSharm** [comment]
Ok, so it works fine when i run "c7n-org run -c accounts.yml -u test.yml -s out" command. But gives error on "custodian validate test.yml", I was trying to validate the policy before c7n-org runs it.

Do you think it could be fixed for custodian validate as well?

policies:
  - name: cloudwatch-logs-retention
    resource: log-group
    description: |
      Sets a default retention on a cloudwatch log group where a retention is currently unset.
    filters:
      - type: value
        key: retentionInDays
        value: absent
    actions:
      - type: retention
        days: "{cw_log_retention_days}"

2023-08-02 10:47:28,666: custodian.commands:ERROR Configuration invalid: test.yml
2023-08-02 10:47:28,666: custodian.commands:ERROR             Error on policy:cloudwatch-logs-retention resource:log-group
'{numofdays}' is not of type 'integer'

            Failed validating 'type' in schema[12]['properties']['days']:
                {'type': 'integer'}

            On instance['days']:
                '{numofdays}'
2023-08-02 10:47:28,666: custodian.commands:ERROR cloudwatch-logs-retention


(custodian) / # which c7n-org
/custodian/bin/c7n-org
(custodian) / # which custodian
/custodian/bin/custodian
(custodian) / # pip3 freeze | grep c7n-org
c7n-org==0.6.28

**Comment by kapilt** [comment]
noted, to fix for custodian validate, we would need to also passing in the c7n-org vars file there, and it would only use top level variables, and additional option / consideration is adding a validate flag to c7n-org

**Comment by mcripps9** [comment]
I just ran into this exact same issue.  @kapilt / @ajkerrigan - what do you think about having custodian validate substitute something like "0" for any user-defined variables, as it is both a string and an integer?  It may not be a perfect solution, but it would cover this rather important use case.

**Comment by ajkerrigan** [comment]
> I just ran into this exact same issue. @kapilt / @ajkerrigan - what do you think about having custodian validate substitute something like "0" for any user-defined variables, as it is both a string and an integer? It may not be a perfect solution, but it would cover this rather important use case.

While I don't think a stringified number will help here, I do think we can try to explicitly pass non-string type checks when we have a variable reference. I'm taking a shot at that in #9237, though I'm sure there are some wrinkles left to iron out.

**Comment by ajkerrigan** [comment]
Closed by #9237
